tests/unit_tests/data_steward/utils/pipeline_logging_test.py

```diff
@@ -69,6 +96,33 @@ def test_create_logger(self, mock_get_logger, mock_file_handler,
             mock_stream_handler.return_value)
         mock_get_logger.assert_called_with(self.log_file_list[1])
 
+    def test_get_logger(self):
+        file_emit = FakeEmit()
+        stderr_emit = FakeEmit()
+        logger_name = 'pipeline_logging_test'
+        mock_open = mock.patch('logging.FileHandler._open',
+                               mock.mock_open(),
+                               create=True)
+        mock_open.start()
+        with mock.patch('logging.FileHandler.emit', new=file_emit):
+            with mock.patch('logging.StreamHandler.emit', new=stderr_emit):
+                # calling multiple times results in the same logger instance
+                self.assertEqual(id(get_logger(logger_name)),
+                                 id(get_logger(logger_name)))
+                logger = get_logger(logger_name)
+                self.assertEqual(logger_name, logger.name)
+                logger.debug('debug message')
+                # debug messages are logged to file
+                self.assertEqual(file_emit.messages, ['debug message'])
+                # debug messages are NOT logged to stderr
+                self.assertEqual(0, stderr_emit.call_count)
+
+                logger.info('info message')
+                self.assertEqual(file_emit.messages,
+                                 ['debug message', 'info message'])
+                self.assertListEqual(stderr_emit.messages, ['info message'])
```

Comment:
What does this give us that using the [standard context manager](https://docs.python.org/3.7/library/unittest.html#unittest.TestCase.assertLogs) does not give us?

Reply:
The context manager has a lot of the same, but I couldn't get it to tell me which handlers log records get routed to. I wanted to ensure that debugs go to the file handler as it seemed this could be perturbed by e.g. prior logging initialization. Please let me know if you have recommendations- I googled the heck out of this and would definitely prefer tried-and-true approaches over this one.